src/libYARP_dev/include/yarp/dev/IDepthSensor.h

```diff
@@ -0,0 +1,225 @@
+// -*- mode:C++; tab-width:4; c-basic-offset:4; indent-tabs-mode:nil -*-
+
+/*
+ * Copyright (C) 2016 iCub Facility - Istituto Italiano di Tecnologia
+ * Author:  Alberto Cardellino
+ * email:   alberto.cardellino@iit.it
+ * website: www.robotcub.org
+ * Permission is granted to copy, distribute, and/or modify this program
+ * under the terms of the GNU General Public License, version 2 or any
+ * later version published by the Free Software Foundation.
+ *
+ * A copy of the license can be found at
+ * http://www.robotcub.org/icub/license/gpl.txt
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details
+ */
+
+#ifndef YARP_DEPTHSENSOR_INTERFACE
+#define YARP_DEPTHSENSOR_INTERFACE
+
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/sig/Matrix.h>
+#include <yarp/sig/Image.h>
+#include <yarp/os/Stamp.h>
+
+// Common usage vocabs (also defined in ControlBoardInterfaces.h)
+#define VOCAB_SET VOCAB3('s','e','t')
+#define VOCAB_GET VOCAB3('g','e','t')
+#define VOCAB_IS  VOCAB2('i','s')
+#define VOCAB_OK  VOCAB2('o','k')
+#define VOCAB_FAILED VOCAB4('f','a','i','l')
+
+#define VOCAB_IDEPTH_SENSOR         VOCAB4('i','d','p','t')
+#define VOCAB_SENSOR_INFO           VOCAB4('i','n','f','o')
+#define VOCAB_SENSOR_STATUS         VOCAB4('s','t','a','t')
+#define VOCAB_DISTANCE_RANGE        VOCAB4('r','a','n','g')
+#define VOCAB_HSCAN_LIMIT           VOCAB4('h','l','i','m')
+#define VOCAB_VSCAN_LIMIT           VOCAB4('v','l','i','m')
+#define VOCAB_DATA_SIZE             VOCAB4('s','i','z','e')
+#define VOCAB_HV_RESOLUTION         VOCAB3('r','e','s')
+#define VOCAB_SCAN_RATE             VOCAB4('r','a','t','e')
+
+namespace yarp {
+    namespace dev {
+        class IDepthSensor;
+    }
+}
+
+/**
+ * @ingroup dev_iface_other
+ *
+ * A generic interface for depth sensor device
+ * This implementation is targeted for depth cameras like
+ * kinect devices.
+ * For linear scanner, the IRangefinder2D interface shall be used instead.
+ *
+ * This interface could be extended in the future to take
+ * into account other devices like planar lasers or single point
+ * range measurement devices like ultrasound systems, infrared proximity
+ * sensors and so on, but not yet at this time.
+ *
+ * Laser or light devices can easily fit into this interface while
+ * sonar devices doesn't, so a different interface/device shall be used instead.
+ */
+
+class YARP_dev_API yarp::dev::IDepthSensor
+{
+public:
+    enum DepthSensor_status
+    {
+        DEPTHSENSOR_NOT_READY        = 0,
+        DEPTHSENSOR_OK_STANBY        = 1,
+        DEPTHSENSOR_OK_IN_USE        = 2,
+        DEPTHSENSOR_GENERAL_ERROR    = 3,
+        DEPTHSENSOR_TIMEOUT          = 4
+    };
+
+    enum VerboseLevel
```

Comment:
There are 2 reasons for them:
1  all Y** functions are always computed, even if not printed at computation cost
2  a part from yTrace, all are always printed, while it would be nice to add a --verbose option when running the module to enable more info for debugging but leave it quiet when everything is ok. There is no way to do this right now